manual/01-intro/Immutability.md

```diff
@@ -1,4 +1,168 @@
+mutate the underlying object.  Ramda's equivalents just return new lists:
+
+```js
+var a1 = [1, 2, 3];
+var b1 = a1.reverse(); //=> [3, 2, 1]
+a1; //=> [3, 2, 1]
+
+var a2 = [1, 2, 3];
+var b2 = R.reverse(a2); //=> [3, 2, 1]
+a2; //=> [1, 2, 3]
+
+// and similarly for `sort`
+```
+
+We won't discuss in depth all the reasons that immutability is desirable.
+That's easily available [on the web][go].  But there are many important
+reasons having to do with how well anyone can understand our code, how
+well it stands up to parallelization, how many fewer ways in which it can
+actually fail.
+
+We also will not talk about how you might make more of our own data
+immutable (although in [Chapter 10][im], we do discuss how Ramda will
+integrate with some immutable data libraries but not others.)  But we do
+want to stress that _none_ of Ramda's functions mutate user data.
+
+For those used to functional programming, we should point out that this 
+is not done through deeply recursive algorithms.  We have plenty of 
+internal mutation.  But at our function boundaries, everything can be 
+seen as immutable.
+
+The basic point is simple: Don't expect Ramda to mutate your data for
+you:
+
+```js
+var numbers = [8, 6, 7, 5, 3, 0];
+var jenny = R.append(9, numbers);
+numbers; //=> [8, 6, 7, 5, 3, 0]
+jenny; //=> [8, 6, 7, 5, 3, 0, 9]
+```
+
+Ramda does not take any special pains to prevent you from mutating
+your own data when using its functions.  For intance you could build 
+up an ouput object by continually mutating the accumulator passed to
+[`reduce`][re].  Ramda would not prevent you:
+
+```js
+var states = [
+    {symbol: 'CT', name: 'Connecticut', pop: 3574097},
+    {symbol: 'ME', name: 'Maine', pop: 1328361},
+    {symbol: 'MA', name: 'Massachusetts', pop: 6547629},
+    {symbol: 'NH', name: 'New Hampshire', pop: 1316470},
+    {symbol: 'RI', name: 'Rhode Island', pop: 1052567},
+    {symbol: 'VT', name: 'Vermont', pop: 623741},
+];
+
+var notSmall = (state) => state.pop >= 2000000;
+
+R.reduce((accum, state) => {
+    if (notSmall(state)) {
+        accum[state.symbol] = state.pop;
+    }
+    return accum;
+}, {}, states); //=> {"CT": 3574097, "MA": 6547629}
+```
+
+Notice the continual mutation of the accumulator in this sample.  This
+is not in any way prohibited by Ramda.  But it is perhaps not the most
+idiomatic way to use Ramda.  The `assoc` we used earlier would cover 
+the same case without introducing such continual mutation:
+
+```js
+R.reduce((accum, state) => notSmall(state) ? 
+                           R.assoc(state.symbol, state.pop, accum) : 
+                           accum,
+{}, states),  //=> {"CT": 3574097, "MA": 6547629}
+```
+
+And, unless it caused performance concerns, many Ramda users would
+prefer to do this by piping the data through a few simpler 
+transformations:
+
+```js
+R.pipe(
+    R.filter(notSmall), 
+    R.map((state) => [state.symbol, state.pop]), 
```

Comment:
Should we use `R.props(['symbol', 'pop'])` here in place of the lambda? It seems idiomatic to me, but perhaps you were keen to avoid introducing too many functions at once.